code/WorkInProgress/nadir_antenna.dm

```diff
@@ -56,6 +58,31 @@ var/global/obj/machinery/communications_dish/transception/transception_array
 				src.failsafe_active = FALSE
 			else
 				src.attempt_restart()
+		else if(length(direct_queue) && primed && !src.is_transceiving)
+			var/obj/queued_item = pick(direct_queue)
+			for_by_tcl(transc_pad, /obj/machinery/transception_pad)
+				if(transc_pad.is_transceiving)
+					continue
+				var/datum/powernet/pad_powernet = transc_pad.get_direct_powernet()
+				if(!pad_powernet)
+					continue
+				var/turf/receive_turf = get_turf(transc_pad)
+				var/obstructed = FALSE
+				if(length(receive_turf.contents) < 10) //move on to the next pad if there is excessive clutter or dense object
+					for(var/atom/movable/O in receive_turf)
+						if(istype(O,/obj))
+							if(O.density)
+								obstructed = TRUE
+								break
+				else
+					obstructed = TRUE
+				if(obstructed)
+					continue
+				var/pad_netnum = pad_powernet.number
+				if(src.can_transceive(pad_netnum))
```

Comment:
looks like this proc returns a number of different failure values, only one of which is falsy, intentional?

Reply:
Good catch. Fixing momentarily.

Reply:
Fixed in 9b1df8f. Checked that it still operates post-change by spawning in successfully with the Mail Order trait.